lib/runtime-core/src/typed_func.rs

```diff
@@ -246,6 +297,97 @@ where
     }
 }
 
+impl<'a> ErasedFunc<'a> {
+    /// Creates a polymorphic function.
+    #[allow(unused_variables)]
+    #[cfg(all(unix, target_arch = "x86_64"))]
```

Comment:
Can we have another implementation of `new_polymorphic` that is for not `x86_64`?

`#[cfg(not(all(unix, target_arch = "x86_64")))]` that raises a `compile_error!("Polymorphic functions are not implemented for this architecture")`

Reply:
Making it a compile error will prevent `runtime_core` to build on non-`x86_64` architectures, while making it a runtime error doesn't make a lot of sense since external code should be aware of the unsupported status at compile time. Therefore I think leaving it "missing" for now would be fine.